burrito/cmd/serve.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright © 2021 NAME HERE <EMAIL ADDRESS>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"burrito/utils"
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+// serveCmd represents the serve command
+var serveCmd = &cobra.Command{
+	Use:   "serve",
+	Short: "Serve a URL Endpoint hosting the downloaded executables",
+	Long: `
+	Read the configuration file and serve the files, the files will be served. 
+	under {fsDir}/{componentName}/{resource} For example:
+	"containerd_url": "http://localhost:3000/files/containerd/containerd-1.5.2-windows-amd64.tar.gz"
+	
+	Defaults: 
+	- port: 80`,
+	Run: func(cmd *cobra.Command, args []string) {
+		var mc utils.BurritoConfig
+		if err := viper.Unmarshal(&mc); err != nil {
+			fmt.Println(err)
```

Comment:
see comment about moving this to the init method but if this fails you wont have an httproot, you sure you dont just want to fatal or panic here?